### PR TITLE
Fix for unauthorized access and push notification 

### PR DESCRIPTION
**Pull request by kunall17**
ddec1c2 aims to fix the Unauthorised errors, its not a good thing for a server
device-2017-08-16-230805

This prevents to running of the event loop if the apiKey is '', which can only happen if the user logouts! 

255448c: Due to some refactoring the savePushToken wasn't working fine, now it will

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/account-info/LogoutButton.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ class LogoutButton extends PureComponent {
   shutdownPUSH = async () => {
     const { auth, actions, pushToken } = this.props;
     if (pushToken !== '') {
-      await unregisterPush(auth, pushToken);
+      try {
+        await unregisterPush(auth, pushToken);
+      } catch (e) {
+        console.log('failed to unregister Push token', e); // eslint-disable-line
```
Comment (borisyankov) [inline src/account-info/LogoutButton.js]: Is this enough? Are you using it to silence the exception or?
When does that happen?

File: src/realm/realmActions.js [hunk 2]
```diff
@@ -77,11 +77,15 @@ export const deleteTokenPush = (): Action => ({
   type: DELETE_TOKEN_PUSH,
 });
 
-export const saveTokenPush = (pushToken: string): Action => ({
+const saveTokenPush = (pushToken: string) => ({
```
Comment (borisyankov) [inline src/realm/realmActions.js]: Did you intentionally remove the `export`?

Comment (kunall17) [inline src/realm/realmActions.js]: Yup, the saveTokenPush is not being used any where else

File: src/account-info/LogoutButton.js (hunk 1 above)
Comment (kunall17) [inline src/account-info/LogoutButton.js]: Ideally we should hook up the sentry to log these error events, I'm not sure how to do it know, hence a log statement! 

It shouldn't happen though, the only param is the pushToken and this function is called the token is not ''

Comment (borisyankov) [inline src/account-info/LogoutButton.js]: OK, good that you have considered it. Indeed we can hook up Sentry later.

**Pull request merged**